### PR TITLE
feat: [refractionPOINT/tracking#3025] Add new "limacharlie users invite" action for inviting a user

### PR DESCRIPTION
**Pull request by tomaz-lc**
## Description

This pull request adds new `limacharlie users invite --email=<email>` CLI action for inviting a user.

## Example Usage

```bash
# Invalid usage - 1
$ limacharlie users
usage: limacharlie user [-h] {invite} ...
limacharlie user: error: the following arguments are required: action

# Invalid usage - 2
$ limacharlie users invite
Error: Please provide either --email or --file option.

# Invalid usage - 3
$ limacharlie users invite --email=tomaz@tomaz.me --file=1.txt
Error: --email and --file are mutually exclusive, please provide only one.

# Success - already invited
$ limacharlie users invite --email=tomaz@tomaz.me
User with email tomaz@tomaz.me already exists / has already been invited.

# Success - invite sent
$ limacharlie users invite --email=tomaz+test1@tomaz.me
User with email tomaz+test1@tomaz.me has been invited.

# Invite multiple users using comma delimited --email flag
$ limacharlie users invite --email=tomaz+test1@tomaz.me,tomaz+test2@tomaz.me,tomaz+test3@tomaz.me
Inviting user with email tomaz+test1@tomaz.me.
User with email tomaz+test1@tomaz.me already exists / has already been invited.
Inviting user with email tomaz+test2@tomaz.me.
User with email tomaz+test2@tomaz.me already exists / has already been invited.
Inviting user with email tomaz+test3@tomaz.me.
User with email tomaz+test3@tomaz.me already exists / has already been invited.

# Invite multiple users from new line delimited text file
$ cat 1.txt 
tomaz+test1@tomaz.me
tomaz+test2@tomaz.me
tomaz+test12@tomaz.me

$ limacharlie users invite --file=1.txt
Found 3 emails in the file.
Inviting user with email tomaz+test1@tomaz.me.
User with email tomaz+test1@tomaz.me already exists / has already been invited.
Inviting user with email tomaz+test2@tomaz.me.
User with email tomaz+test2@tomaz.me already exists / has already been invited.
Inviting user with email tomaz+test12@tomaz.me.
User with email tomaz+test12@tomaz.me has been invited.
```

## Tests

I wanted to add some basic tests, but I noticed our tests work with the actual API (aka they are end to end tests which use real credentials from a testing account).

Since this functionality results in sending out an email, I'm not sure if it's a best idea to generate partially random email for each test run (e.g. `tomaz+test-run-XXX@limacharlie.com`, it would need to be valid one to prevent bounces) - I want to make sure this doesn't negatively affect our reputation with our email provider.

At the very least, I could add a test which verifies edge case when a user already exists since this doesn't result in any email being sent out.

## TODO

- [x] Decide on the singular vs plural action name - `user` vs `users`. I first used `user`, but then I noticed most commands use plural form so I went with the same option for consistency reasons.
- [x] Decide how to handle tests
- [x] Documentation on how to use this command

## Type of change

- [x] New feature (non-breaking change that adds functionality)

## Related issues

Fix refractionPOINT/tracking#3025

### CODE REVIEW

File: limacharlie/User.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        if exists:
+            print("User with email %s already exists / has already been invited." % (email))
+        else:
+            print("User with email %s has been invited." % (email))
+
+def main( sourceArgs = None ):
+    import argparse
+
+    parser = argparse.ArgumentParser( prog = 'limacharlie user' )
+    # TODO: We should correctly use argparse subparsers everywhere or use something like click library...
```
Comment (tomaz-lc) [inline limacharlie/User.py:40]: I noticed our existing CLI abstraction leaves a lot to be desired.

We don't (correctly) use sub-parsers nor some library like `click`.

This means that the most existing commands don't work as you would expect them out of the box with standard Python abstractions (error handling, bash completions and many other things).

Of course fixing / changing all of that would be a large effort and we would of course also need to keep it backward compatible which would not be completely trivial.

**Review by maximelb — APPROVED** [review]

**Comment by dzimine-lc** [comment]
I'd like to have a batch invite functionality at the CLI level (not API) - give a comma-separated list of emails and/or a file with comma-sep or new line emails. 
Yes one can do it with a script but given that batch-add is a prime use case I move we want it covered here.

**Comment by tomaz-lc** [comment]
@dzimine-lc ACK, I will add it - I will think a bit about the API and check consistency withthe  existing commands and decide what's the best course of action (e.g. `limacharlie users invite --email=foo,bar,baz` vs `limacharlie users invite --email email1 --email email2`  vs `limacharlie users invite-multi foo,bar,baz`).

**Comment by tomaz-lc** [comment]
@dzimine-lc I pushed a change to support multiple users at once - a7e2134500dac9218cab6b3a7be993cb13adb322.

It supports two notations:

1.  Multiple emails specified in comma delimited format via `--email` flag, e.g. `--email=test1@example.com,test2@example.com`
2. Via new line delimited entries from a file on disk - e.g. `--file=emails_to_invite.txt`

See the PR description for updated usage examples.

Feel free to test the ergonomics / review it, but it's not fully finished yet. I need to pull in changes from #156 and then add some basic unit tests.

**Comment by tomaz-lc** [comment]
I've pushed a fix for a regression introduced in #156 (https://github.com/refractionPOINT/python-limacharlie/pull/156#issuecomment-2671207282) and also added tests + some other improvements.

The PR is now ready for a review.

@dzimine-lc You can review the output in the PR description or test it locally yourself by doing something like:

```bash
source venv-test-lc-cli/bin/activate
python3 -m venv venv-test-lc-cli
pip install git+https://github.com/refractionPOINT/python-limacharlie@feat/3025/new-user-invite-action#egg=limacharlie
```

Also keep in mind that for this operation, you need to use a user scoped API key. More info at https://docs.limacharlie.io/apidocs/introduction#getting-a-jwt.

File: limacharlie/__main__.py [hunk 2]
```diff
@@ -49,7 +56,6 @@ def cli(args):
             oid = raw_input( 'Enter your Organization ID (UUID): ' ) # noqa
         else:
             oid = input( 'Enter your Organization ID (UUID): ' )
-        print(oid)
```
Comment (tomaz-lc) [inline limacharlie/__main__.py:52]: This accidentaly slipped through via #156.

**Review by maximelb — APPROVED** [review]
